data-prepper-plugins/aws-lambda/src/main/java/org/opensearch/dataprepper/plugins/lambda/processor/LambdaProcessor.java

```diff
@@ -162,104 +165,238 @@ public Collection<Record<Event>> doExecute(Collection<Record<Event>> records) {
         try {
+            // Current Assumption: Lambda always returns json array.
+            // TODO: add support for jsonObject/string/object return from lambda
+            boolean isJsonCodec = responseCodec instanceof JsonInputCodec;
+
+            //Get payload and convert to response codec
             SdkBytes payload = lambdaResponse.payload();
-            if (payload != null) {
-                String payloadJsonString = payload.asString(StandardCharsets.UTF_8);
-
-                JsonNode jsonNode = null;
-                try {
-                    jsonNode = objectMapper.readTree(payloadJsonString);
-                } catch (JsonParseException e) {
-                    throw new RuntimeException("payload output is not json formatted");
+
+            // Handle null or empty payload
+            if (payload == null || payload.asByteArray() == null || payload.asByteArray().length == 0) {
+                LOG.error(NOISY, "Lambda response payload is null or empty");
+                throw new RuntimeException("Lambda response payload is null or empty");
+            }
+
+            // Record payload sizes
+            requestPayloadMetric.set(flushedBuffer.getPayloadRequestSize());
+            responsePayloadMetric.set(payload.asByteArray().length);
+
+            LOG.debug("Response payload:{}",payload.asUtf8String());
+            InputStream inputStream = new ByteArrayInputStream(payload.asByteArray());
+            List<Event> parsedEvents = new ArrayList<>();
+            responseCodec.parse(inputStream, record -> {
+                Event event = record.getData();
+                parsedEvents.add(event);
+            });
+            if(parsedEvents.isEmpty()){
+                LOG.error("Parsing response failed for event, response:{}", payload.asUtf8String());
+                throw new RuntimeException("Response parsing failed");
+            }
+
+            List<Record<Event>> originalRecords = flushedBuffer.getRecords();
+
+            LOG.debug("Parsed Event Size:{}, FlushedBuffer eventCount:{}, FlushedBuffer size:{}",parsedEvents.size(),flushedBuffer.getEventCount(),flushedBuffer.getSize());
+            // Check if the response is a JSON array and the codec is JSON
+            if (isJsonCodec) {
+                if (parsedEvents.size() == flushedBuffer.getEventCount()) {
+                    LOG.debug("Parsed event size from lambda = event count sent to lambda");
+                    for (int i = 0; i < parsedEvents.size(); i++) {
+                        Event responseEvent = parsedEvents.get(i);
+                        Event originalEvent = originalRecords.get(i).getData();
+
+                        // Delete the original event's data
+                        originalEvent.clear();
+
+                        // Replace data from responseEvent into originalEvent
+                        Map<String, Object> responseData = responseEvent.toMap();
```

Comment:
This logic is handling strict and aggregates the same. If the behavior is aggregate, then we should always create new events. It doesn't matter if the size is the same or not.